src/System.Runtime.Loader/tests/DefaultContext/DefaultLoadContextTest.cs

```diff
@@ -58,24 +58,43 @@ public static void LoadInDefaultContext()
         {
             Init();
 
-           // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
-           var assemblyName = "System.Runtime.Loader.Noop.Assembly";
+            // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
+            var assemblyNameStr = "System.Runtime.Loader.Noop.Assembly";
+            var assemblyName = new AssemblyName(assemblyNameStr);
 
+            // By default, the assembly should not be found in DefaultContext at all
+            Assert.Throws(typeof(FileNotFoundException), () => Assembly.Load(assemblyName));
+
+            // Now, wireup the Resolving event to locate the assembly
             AssemblyLoadContext.Default.Resolving += ResolveAssembly;
-            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(new AssemblyName(assemblyName));
+            
+            // This will invoke the resolution via VM requiring to bind using the TPA binder
+            var assemblyExpectedFromLoad = Assembly.Load(assemblyName);
+
+            // We should have successfully loaded the assembly in default context.
+            Assert.NotNull(assemblyExpectedFromLoad);
+
+            // And make sure the simple name matches
+            Assert.Equal(assemblyNameStr, assemblyExpectedFromLoad.GetName().Name);
+
+            // This will resolve the assembly via event invocation.
+            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
             
             // We should have successfully loaded the assembly in default context.
             Assert.NotNull(assemblyExpected);
 
+            // What we got via Assembly.Load and LoadFromAssemblyName should be the same
+            Assert.Equal(assemblyExpected, assemblyExpectedFromLoad);
+
             // And make sure the simple name matches
-            Assert.Equal(assemblyExpected.GetName().Name, assemblyName);
+            Assert.Equal(assemblyExpected.GetName().Name, assemblyNameStr);
 
             // Unwire the Resolving event.
             AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
```

Comment:
> Yes.

At least in this file, I only see one test unsubscribing from Resolving, and it's the case where the assembly was already previously loaded so Resolving doesn't matter.  Do we have one elsewhere that validates loading fails after Resolving is unsubscribed from if the assembly wasn't previously loaded?

> First one to return wins - there is no order defined/expected.

Do we have a test locking in this behavior?  Even if it's not guaranteed/documented, a test would help us to know if the behavior ever changes in this regard.

> No - but this will be good to add. Will do that.

Ok, thanks.

> This will be taken care of by the test case I add for the previous case.

Ok.